### PR TITLE
Fixes #32 Improve exception handling

### LINKED ISSUE #32: VSSolution Parser throws up on Missing EndProject in Solution, but Error is clouded by the Task.
Improve the error reporting in Tasks in general and for the VsSolutionParser in particular

### PR DESCRIPTION
**Pull request by FreeApophis**

### CODE REVIEW

**Comment by FreeApophis** [comment]
Die ersten 3 commits gehören zu PR #48 - ich brauch PR #48 als Basis für FormatException.

Also PR #48 zuerst mergen.

**Comment by FreeApophis** [comment]
Beispiel aus Bug #32 

Exception in module: MetaGeneratorProjectPlugin
Expecting Messerli.VsSolution.Token.EndProjectToken but got Project at Line 43 Column 1.

**Review by dngroth — DISMISSED** [review]

File: VsSolution/Parser/SolutionParser.cs [hunk 1]
```diff
@@ -28,8 +28,8 @@ public static SolutionParser Create()
         {
             // Create the object tree without DI Framework
             var lexerRules = new LexerRules();
-            var tokenizer = new Tokenizer(lexerRules, s => new LexerReader(s));
-            var tokenWalker = new TokenWalker(tokenizer, () => new EpsilonToken());
+            var tokenizer = new Tokenizer(lexerRules, s => new LexerReader(s), l => new LinePositionCalculator(l));
```
Comment (janhohenheim) [inline VsSolution/Parser/SolutionParser.cs]: The number of parameters is starting to warrant explicit parameter identifiers.

File: MetaGenerator/ExceptionFormatter.cs [hunk 2]
```diff
@@ -21,12 +21,22 @@ public ExceptionFormatter(IConsoleWriter consoleWriter, IGlobalOptions globalOpt
...
+                    FormatException(innerException);
+                }
+            }
+            else
             {
-                _consoleWriter.WriteLine(exception.StackTrace);
+                _consoleWriter.WriteLine($"Exception in module: {Module()}".Pastel(Color.LightGoldenrodYellow));
+                _consoleWriter.WriteLine(exception.Message.Pastel(Color.OrangeRed));
+
+                if (_globalOptions.Verbose && exception.StackTrace is { })
```
Comment (janhohenheim) [inline MetaGenerator/ExceptionFormatter.cs]: remove (slight) code duplication by binding `exception.StackTrace` to a variable like so: `exception.StackTrace is { } stackTrace`

File: MetaGenerator/ExceptionFormatter.cs [hunk 3]
```diff
@@ -21,12 +21,22 @@ public ExceptionFormatter(IConsoleWriter consoleWriter, IGlobalOptions globalOpt
 
         public void FormatException(Exception exception)
         {
-            _consoleWriter.WriteLine($"Exception in module: {Module()}");
-            _consoleWriter.WriteLine(exception.Message.Pastel(Color.OrangeRed));
-
-            if (_globalOptions.Verbose && exception.StackTrace is { })
+            if (exception is AggregateException aggregateException)
```
Comment (janhohenheim) [inline MetaGenerator/ExceptionFormatter.cs:25]: Move the positive and negative branches into own functions.

**Review by janhohenheim — CHANGES_REQUESTED** [review]

**Review by janhohenheim — APPROVED** [review]

**Pull request merged**